core/services/webhook/external_initiator_manager.go

```diff
@@ -85,14 +83,52 @@ func (m externalInitiatorManager) Notify(webhookSpecID int32) error {
 }
 
 func (m externalInitiatorManager) Load(webhookSpecID int32) (eiWebhookSpecs []job.ExternalInitiatorWebhookSpec, jobID uuid.UUID, err error) {
-	row := m.db.Raw("SELECT external_job_id FROM jobs WHERE webhook_spec_id = ?", webhookSpecID).Row()
-	err = multierr.Combine(
-		errors.Wrapf(row.Scan(&jobID), "failed to load job ID from job for webhook spec with ID %d", webhookSpecID),
-		errors.Wrapf(m.db.Where("webhook_spec_id = ?", webhookSpecID).Preload("ExternalInitiator").Find(&eiWebhookSpecs).Error, "failed to load external_initiator_webhook_specs for webhook_spec_id %d", webhookSpecID),
-	)
+	if err = m.db.Get(&jobID, "SELECT external_job_id FROM jobs WHERE webhook_spec_id = $1", webhookSpecID); err != nil {
+		if err = errors.Wrapf(err, "failed to load job ID from job for webhook spec with ID %d", webhookSpecID); err != nil {
+			return
+		}
+	}
+	if err = m.db.Select(&eiWebhookSpecs, "SELECT * FROM external_initiator_webhook_specs WHERE external_initiator_webhook_specs.webhook_spec_id = $1", webhookSpecID); err != nil {
+		if err = errors.Wrapf(err, "failed to load external_initiator_webhook_specs for webhook_spec_id %d", webhookSpecID); err != nil {
+			return
+		}
+	}
+	if err = m.preloadExternalInitiator(eiWebhookSpecs); err != nil {
+		if err = errors.Wrapf(err, "failed to preload ExternalInitiator for webhook_spec_id %d", webhookSpecID); err != nil {
+			return
+		}
+	}
 	return
 }
 
+func (m externalInitiatorManager) preloadExternalInitiator(txs []job.ExternalInitiatorWebhookSpec) error {
+	var ids []int64
+	for _, tx := range txs {
+		ids = append(ids, tx.ExternalInitiatorID)
+	}
+	if len(ids) == 0 {
+		return nil
+	}
+	query, args, err := sqlx.In(`SELECT * FROM external_initiators WHERE external_initiators.id IN (?);`, ids)
```

Comment:
Personally I prefer to use `WHERE xxx = ANY($1)` because it avoids the extra rebinding - you can pass the array directly